### PR TITLE
Fix for #118: StringType defaults to TEXT which is varchar(256)

### PR DESCRIPTION
**Pull request by eprochasson**
also fixed a bit of documentation about compound key.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -352,10 +352,32 @@ To support larger columns, you can use the `maxlength` column metadata field to
...
+columnWidthMap.foreach(m => {
+  df = df.withColumn(m._1, df(m._1).as(m._1, new MetadataBuilder().putLong("maxlength", m._2).build()))
+})
+
+df.write
+  .format("com.databricks.spark.redshift")
+  .option("url", jdbcURL)
+  .option("tempdir", s3TempDirectory)
+  .option("dbtable", sessionTable)
+  .mode(SaveMode.Overwrite)
```
Comment (JoshRosen) [inline README.md]: Suuuuuper minor nit, but I'm slightly wary of putting this into example code that a user might verbatim copy-and-paste. Probably better to omit the use of a more powerful SaveMode.

File: README.md [hunk 2]
```diff
@@ -352,10 +352,32 @@ To support larger columns, you can use the `maxlength` column metadata field to
...
+  "column2" -> 11,
+  "column3" -> 12,
+  "column4" -> 13,
+  "column5" -> 14,
+)
+
+var df // the dataframe you'll want to write to redshift
+
+// Apply each column metadata customization
+columnWidthMap.foreach(m => {
```
Comment (JoshRosen) [inline README.md]: Style nit: you could write `columnWidthMap.foreach { m =>` in order to omit an extra layer of braces.

File: README.md [hunk 3]
```diff
@@ -352,10 +352,32 @@ To support larger columns, you can use the `maxlength` column metadata field to
...
+  df = df.withColumn(m._1, df(m._1).as(m._1, new MetadataBuilder().putLong("maxlength", m._2).build()))
```
Comment (JoshRosen) [inline README.md]: This line is kinda long, so you could split it into two by defining `val metadata = ...` on the previous line, then using that metadata when re-assigning to `df`.

File: README.md [hunk 4]
```diff
@@ -352,10 +352,32 @@ To support larger columns, you can use the `maxlength` column metadata field to
...
+// Specify the custom width of each column
+val columnWidthMap = Map(
+  "column1" -> 10,
+  "column2" -> 11,
+  "column3" -> 12,
+  "column4" -> 13,
+  "column5" -> 14,
+)
+
+var df // the dataframe you'll want to write to redshift
```
Comment (JoshRosen) [inline README.md]: Nit: capitalize Redshift whenever it appears, since it's the name of a product.

File: README.md [hunk 5]
```diff
@@ -352,10 +352,32 @@ To support larger columns, you can use the `maxlength` column metadata field to
 
 Here is an example of updating a column's metadata field in Scala:
```
Comment (JoshRosen) [inline README.md]: "a column's metadata field" -> "multiple columns' metadata fields"?

File: README.md (hunk 2 above)
Comment (JoshRosen) [inline README.md]: Also, could write `columnWidthMap.foreach { case (colName, length) =>` to avoid the `_1`s.

File: README.md [hunk 6]
```diff
@@ -294,9 +294,9 @@ must also set a distribution key with the <tt>distkey</tt> option.
 
 <p>Examples include:</p>
 <ul>
-    <li><tt>SORTKEY my_sort_column</tt></li>
-    <li><tt>COMPOUND SORTKEY sort_col_1, sort_col_2</tt></li>
-    <li><tt>INTERLEAVED SORTKEY sort_col_1, sort_col_2</tt></li>
+    <li><tt>SORTKEY(my_sort_column)</tt></li>
```
Comment (JoshRosen) [inline README.md]: Good catch (I never tried this feature myself since it predates my work on this project).

**Comment by JoshRosen** [comment]
Thanks for submitting this PR. I left a bunch of nitpicky comments because I'm kind of a perfectionist; if you don't have time to address them, I can take care of them myself on merge tomorrow.

**Comment by eprochasson** [comment]
Ha! Neh I'll do it.

**Comment by eprochasson** [comment]
I made the example slightly less abstract too.

File: README.md [hunk 7]
```diff
@@ -350,12 +350,34 @@ When creating Redshift tables, `spark-redshift`'s default behavior is to create
...
+  "language_code" -> 2,
+  "country_code" -> 2,
+  "url" -> 2083
+)
+
+var df = ... // the dataframe you'll want to write to Redshift
+
+// Apply each column metadata customization
+columnLengthMap.foreach(
+  case (colName, length) => {
```
Comment (JoshRosen) [inline README.md]: You can still remove one level of braces, but I'll do this myself right now on merge.

**Comment by JoshRosen** [comment]
Merged. Thanks!

File: README.md [hunk 8]
```diff
@@ -350,12 +350,34 @@ When creating Redshift tables, `spark-redshift`'s default behavior is to create
...
+    val metadata = new MetadataBuilder().putLong("maxlength", length).build())
```
Comment (JoshRosen) [inline README.md]: Noticed that there was also an extra `)` here, so I fixed that as well.

**Pull request closed without merging**